packages/trader/src/AppV2/Components/ContractCard/contract-card-list.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import clsx from 'clsx';
+import { getContractPath } from '@deriv/shared';
+import { TPortfolioPosition } from '@deriv/stores/types';
+import { TClosedPosition } from 'AppV2/Containers/Positions/positions-content';
+import { TRootStore } from 'Types';
+import ContractCard from './contract-card';
+
+export type TContractCardListProps = {
+    currency?: string;
+    hasButtonsDemo?: boolean;
+    onClickCancel?: (contractId: number) => void;
+    onClickSell?: (contractId: number) => void;
+    positions?: (TPortfolioPosition | TClosedPosition)[];
+    setHasButtonsDemo?: React.Dispatch<React.SetStateAction<boolean>>;
+    serverTime?: TRootStore['common']['server_time'];
+};
+
+const ContractCardList = ({
+    hasButtonsDemo,
+    onClickCancel,
+    onClickSell,
+    positions = [],
+    setHasButtonsDemo,
+    ...rest
+}: TContractCardListProps) => {
+    React.useEffect(() => {
+        let demoTimeout: ReturnType<typeof setTimeout>;
+        if (hasButtonsDemo && setHasButtonsDemo) {
+            demoTimeout = setTimeout(() => setHasButtonsDemo(false), 720); // 720 is the length of demo animation
+        }
+        return () => {
+            if (demoTimeout) clearTimeout(demoTimeout);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
You can consider using the `transitionend` event instead of using the timeout - https://developer.mozilla.org/en-US/docs/Web/API/Element/transitionend_event

Reply:
@balakrishna-deriv i could do it if the initial buttons demo was made using transition. However, the demo is made using animation with keyframes.